service/sharddistributor/handler/executor.go

```diff
@@ -95,9 +101,58 @@ func (h *executor) Heartbeat(ctx context.Context, request *types.ExecutorHeartbe
 		return nil, &types.InternalServiceError{Message: fmt.Sprintf("failed to record heartbeat: %v", err)}
 	}
 
+	// emit shard assignment metrics only if shards are assigned in the background
+	// shard assignment in heartbeat doesn't involve any assignment changes happening in the background
+	// thus there was no shard handover and no assignment distribution latency
+	// to measure, so don't need to emit metrics in that case
+	if shardAssignedInBackground {
+		// emits metrics in background to not block the heartbeat response
+		h.emitShardAssignmentMetrics(request.Namespace, heartbeatTime, previousHeartbeat, assignedShards)
```

Comment:
This is not done in the background I think. Did you mean to spawn a go-routine?

Reply:
It used to be in a goroutine due to a db operation, but now it's only cache is used, so there is no need for a goroutine, removed the comment